builtin/providers/docker/resource_docker_container_funcs.go

```diff
@@ -0,0 +1,282 @@
+package docker
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	dc "github.com/fsouza/go-dockerclient"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceDockerContainerCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	client, err := config.NewClient()
+	if err != nil {
+		return fmt.Errorf("Unable to connect to Docker: %s", err)
+	}
```

Comment:
Any specific reason the client can't be initialized in the provider configuration function?